components/merkledb/src/views/metadata.rs

```diff
@@ -340,7 +346,7 @@ impl<T: RawAccess> IndexesPool<T> {
     {
         let len = self.len();
         let metadata = IndexMetadata {
-            identifier: len + 1,
+            identifier: NonZeroU64::new(len + 1).unwrap(),
             // ^-- Identifier should be non-zero to translate to a correct id in `ResolvedAddress`
```

Comment:
Nit / opinion: I feel that the comment is obsolete now since `NonZeroU64::new` is self-explanatory.